scripts/coderabbit_retry/tests/test_coderabbit_retry.py

```
@@ -0,0 +1,375 @@
+    @patch("scripts.coderabbit_retry.coderabbit_retry.process_pr")
+    @patch("scripts.coderabbit_retry.coderabbit_retry.list_eligible_prs")
+    @patch("scripts.coderabbit_retry.coderabbit_retry.Github")
+    @patch.dict("os.environ", {"REPO": "owner/repo", "GH_TOKEN": "fake-token"}, clear=True)
+    def test_isolates_github_exception_per_pr(
+        self,
+        _mock_github_cls: MagicMock,
+        mock_list: MagicMock,
+        mock_process: MagicMock,
+        caplog: pytest.LogCaptureFixture,
+    ) -> None:
+        def process_side_effect(*, repo_name: str, repo: MagicMock, pr_number: int) -> bool:
```

Ruff (0.16.1)

[warning] Unused function argument: `repo_name`

(ARG001)